### PR TITLE
refactor(x/signal): reset all versions via `ResetTally`

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/3276

### CODE REVIEW

**Comment by rootulp** [comment]
Keeping in draft for now. I plan on rebasing after https://github.com/celestiaorg/celestia-app/pull/3290 merges.

File: app/ante/fee_checker.go [hunk 1]
```diff
@@ -16,6 +17,14 @@ const (
 	priorityScalingFactor = 1_000_000
 )
 
+// The purpose of this wrapper is to enable the passing of an additional paramKeeper parameter in
+// ante.NewDeductFeeDecorator whilst still satisfying the ante.TxFeeChecker type.
+func ValidateTxFeeWrapper(paramKeeper params.Keeper) ante.TxFeeChecker {
+	return func(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error) {
+		return ValidateTxFee(ctx, tx, paramKeeper)
+	}
+}
```
Comment (cmwaters) [inline app/ante/fee_checker.go]: 👍

**Review by cmwaters — DISMISSED** [review]
LGTM

The only edge case against doing this is imagine we release v2 and shortly after v3 (before the network has upgraded to v2). Some nodes upgrade to v2 and signal v2. Some upgrade to v3 and signal v3. v2 eventually receives quorum and the upgrade happens. Now the nodes who signalled v3 have had their "votes" wiped and so they must revote if they want to move to v3

**Comment by rootulp** [comment]
I thought about that edge case and I don't think it's possible because [these lines](https://github.com/celestiaorg/celestia-app/blob/1d73e334a48732ba2ef07caf064a7060175381f0/x/signal/keeper.go#L67-L72). In other words, when the app version is 1, the nodes that try to signal for 3 will hit an `ErrInvalidVersion` because they are signaling for a version too far in advance.

**Review by evan-forbes — APPROVED** [review]
LGTM

**Comment by cmwaters** [comment]
> I thought about that edge case and I don't think it's possible because [these lines](https://github.com/celestiaorg/celestia-app/blob/1d73e334a48732ba2ef07caf064a7060175381f0/x/signal/keeper.go#L67-L72). In other words, when the app version is 1, the nodes that try to signal for 3 will hit an ErrInvalidVersion because they are signaling for a version too far in advance.

Makes sense. I always wasn't sure whether we should allow skip versions or not

**Comment by rootulp** [comment]
I'm not sure why golangci-lint is failing on prepare environment. Error looks unrelated to this PR.

**Comment by rootulp** [comment]
I think we can try skipping the pkg cache

**Review by evan-forbes — DISMISSED** [review]

**Review by cmwaters — APPROVED** [review]

**Review by ninabarbakadze — APPROVED** [review]
sorry for the delay. lgtm

**Pull request merged**